scaffolding-chef/lib/scaffolding.sh

```diff
@@ -38,94 +46,193 @@ do_default_unpack() {
   return 0
 }
 
+do_default_build_config() {
+  scaffolding_build_default_toml
+
+  scaffolding_build_config
+  scaffolding_build_config_bootstrap
+  scaffolding_build_config_client
+  scaffolding_build_config_node
+
+  scaffolding_build_hooks
+  scaffolding_build_hook_run
+}
+
 do_default_build_service() {
-  ## Create hooks
-  mkdir -p "$pkg_prefix/hooks"
-  chown 0644 "$pkg_prefix/hooks"
+  return 0
+}
 
-  # Run hook
-  cat << EOF >> "$pkg_prefix/hooks/run"
-#!/bin/sh
+do_default_build() {
+  scaffolding_detect_policy
+  scaffolding_build_policy
+}
 
-export SSL_CERT_FILE="{{pkgPathFor "core/cacerts"}}/ssl/cert.pem"
+do_default_install() {
+  scaffolding_install_policy
+}
 
-cd {{pkg.path}}
+do_default_strip() {
+  return 0
+}
 
-exec 2>&1
-chef-client -z -l {{cfg.log_level}} -c $pkg_svc_config_path/client-config.rb --once
-exec chef-client -z -i {{cfg.interval}} -s {{cfg.splay}} -l {{cfg.log_level}} -c $pkg_svc_config_path/client-config.rb
-EOF
-  chown 0755 "$pkg_prefix/hooks/run"
+scaffolding_detect_policy() {
+  local search_path="$PLAN_CONTEXT"
+
+  while [ "$search_path" != "/" ]; do
```

Comment:
I think that scaffolding should be very opinionated about where its inputs come from.   Walking up the directory tree to the root of the filesystem could lead to unexpected behavior.    I think I'd like to see this check `$PLAN_CONTEXT/policyfiles/` and stop there. If there are other relative paths that it makes sense to check, it should also do that, but we should be very explicit about where the scaffolding will look for the policyfile.

Reply:
In original code it tried to check:
$PLAN_CONTEXT/policyfiles/
$PLAN_CONTEXT/../policyfiles/
$PLAN_CONTEXT/../../policyfiles/

It expects that you  can potentially have a `policyfiles` directory at the root of a repo, but actually plan somewhere deep inside. I thought to change behavior of scaffolding to use `policyfile` explicitly, but it breaks compatibility with previous behavior.

On other hand this method always called from a studio, so it is safe to look directory up until `/`.